xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDecoder.h

```diff
@@ -69,15 +69,135 @@ namespace kodi
 namespace addon
 {
 
+//==============================================================================
+///
+/// \addtogroup cpp_kodi_addon_audiodecoder
+/// @brief \cpp_class{ kodi::addon::CInstanceAudioDecoder }
+/// **Audio decoder add-on instance**
+///
+/// Framework support for binary addons as audio decoder. This class implements a
+/// way to handle special types of audio files.
+///
+/// There is the complete way from loading the source, to outputting the usable
+/// output audio stream.
+///
+/// The addon.xml defines the capabilities that can be used by this addon.
+///
+/// @note With this addon type only the way with multiple instances is possible!
```

Comment:
The option to have multiple instances is possible with audio-decoder add-ons. This is useful, since some playback engines are riddled by global variables, making decoding of multiple streams using the same instance impossible.